src/CampaignAdmin.php

```diff
@@ -324,12 +326,18 @@ protected function getChangeSetResource(ChangeSet $changeSet, $sync = false)
      * Build item resource from a changesetitem
      *
      * @param ChangeSetItem $changeSetItem
-     * @return array
+     * @return array|false
      */
     protected function getChangeSetItemResource(ChangeSetItem $changeSetItem)
     {
         $baseClass = DataObject::getSchema()->baseDataClass($changeSetItem->ObjectClass);
         $baseSingleton = DataObject::singleton($baseClass);
+
+        // Allow items to opt out of being displayed in changesets
+        if ($baseSingleton->config()->get('hide_in_campaigns')) {
+            return false;
```

Comment:
Is this safe? Would `return []` have the same affect without changing the return type?

Reply:
We could do either - I went for false because it's more descriptive, and this method isn't part of our public API so changing its return type doesn't matter as much, but you're probably right that returning an empty array is safer. Will update.